hub/api/dataset.py

```diff
@@ -59,6 +61,8 @@ def __init__(
                 Use this if you want to specify the storage provider object manually instead of using a tag or url to generate it.
             public (bool, optional): Applied only if storage is Hub cloud storage and a new Dataset is being created. Defines if the dataset will have public access.
             token (str, optional): Activeloop token, used for fetching credentials for Hub datasets. This is optional, tokens are normally autogenerated.
+            like (str/Dataset, optional): The full path for the datasets or Dataset object. If provided, will create a new Dataset with the same tesnors and metas.
```

Comment:
Can't we have a static method for this instead in the Dataset object called something like from_template?
I don't like having these 2 as args to the constructor.

Reply:
we don't even need a static method for this, we should have a method on the dataset that returns a new one